### ISSUE TITLE
Erratic Navigation Behavior in Add/Editting Source 

### ISSUE THREAD

**Issue by lasers** [OpenELEC/OpenELEC.tv#2655]
In the setting option, click "Browse" to browse for new share. I see a parent directory option (..) on top and clicked that. I'm not able to escape the blade dialog. Is this designed on purpose to reduce/prevent some kind of back-and-forward behavior on our part?

It seems that almost every other share options works fine with accessing parent directory to go up one level. On Network Filesystem (NFS), I faced the similar behavior where I'm not able to return backward (possibly because of empty directory inside the shared path).

In order to go up one level, you will have to press ESC to exit the blade dialog. Pressing (..) option will either not work or hover the selected line over to the last accessed directory.

TL;DR -- I cannot go back to parent directories and/or exit when accessing (..) option.

Updated. Apologies. Next time I run into an issue, you will get a perfect bug report. :)
OpenELEC Stable - ION x86_64 3.2.1
